application/src/ext/java/org/opentripplanner/ext/emission/internal/csvdata/trip/EmissionAggregator.java

```diff
@@ -0,0 +1,178 @@
+package org.opentripplanner.ext.emission.internal.csvdata.trip;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.opentripplanner.ext.emission.model.TripPatternEmission;
+import org.opentripplanner.framework.error.OtpError;
+import org.opentripplanner.framework.error.WordList;
+import org.opentripplanner.model.plan.Emission;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.site.StopLocation;
+
+/**
+ * This class is responsible for merging duplicates by calculating the average. This can be
+ * problematic, since the rows in the CSV file might not represent the same "volume" of
+ * samples. For example if the feed contains samples for each day a trip runs, but the
+ * the trip is crouded during weekdays and almost empty on weekends, then just adding the
+ * days together and taking the average does not produce the average emission per passenger.
+ */
+class EmissionAggregator {
+
+  private final FeedScopedId tripId;
+  private final List<StopLocation> stops;
+  private final Emission[] emissions;
+  private final int[] counts;
+  private final List<OtpError> issues = new ArrayList<>();
+  private boolean semanticValidationDone = false;
+
+  EmissionAggregator(FeedScopedId tripId, List<StopLocation> stops) {
+    this.tripId = tripId;
+    this.stops = stops;
+
+    int size = stops.size() - 1;
+
+    this.emissions = new Emission[size];
+    Arrays.fill(emissions, Emission.ZERO);
+
+    this.counts = new int[size];
+    Arrays.fill(counts, 0);
+  }
+
+  EmissionAggregator mergeEmissionForleg(TripLegsRow row) {
```

Comment:
Is it the leg or the hop?

Reply:
I will do the leg -> hop refactoring in a separate PR after the 3 pending PRs are merged.